### DISCUSSION TITLE
Enhancement: Send Ctrl-Alt-Del to remote computer

### DISCUSSION THREAD

**Discussion by stevesobol** [immense/Remotely#916] [Ideas]
This is something I need in order to be able to properly support the handful of remote IT clients I have. I've been researching the Win32 `SendSAS` function, and I think I can use it to add this functionality to Remotely, but I was curious if anyone else has already started adding the C-A-D functionality.

**Comment by bitbound** [comment]
There's a Ctrl+Alt+Del button in the menu.

![image](https://github.com/user-attachments/assets/fa7eda95-0189-48a5-a778-b10ed61bf0af)

**Reply by stevesobol** [reply]
Is that a new thing? I don't see it.

**Reply by stevesobol** [reply]
Ahhhh... Remotely must be installed as a service. That makes sense. SendSAS only works with a signed binary or if used from a Windows service. Since I'm planning on using Remotely for unattended access, and will install it as a service on all of the computers I support, we're good. I'm guessing that this option will also be available if I have someone run a signed .exe to create an ad-hoc connection; is this correct?

Thanks.

**Reply by bitbound** [reply]
Signing the binary will not enable the option.  For SendSAS to work from a non-service process, it must have the `UIAccess` flag set in the manifest.  From what I've read, it seems like the flag is only intended to be used by accessibility software.  Requirements are listed [here](https://learn.microsoft.com/en-us/windows/win32/winauto/uiauto-securityoverview#uiaccess-requirements-for-assistive-technology-applications).

I didn't even attempt using that, as I'm guessing there could be repercussions for abusing it.

So the button will only be shown if the agent is installed as a service.  Clicking the button sends the request to the agent, not the streamer in the desktop.
